src/TokenManagers/XboxTokenManager.js

```diff
@@ -68,32 +68,43 @@ class XboxTokenManager {
   }
 
   async verifyTokens (relyingParty) {
+    if (this.forceRefresh) return false
     const ut = await this.getCachedUserToken()
     const xt = await this.getCachedXstsToken(relyingParty)
-    if (!ut || !xt || this.forceRefresh) {
-      return false
-    }
     debug('[xbl] have user, xsts', ut, xt)
-    if (ut.valid && xt.valid) {
+    if (xt.valid) return true
+    if (!ut.valid) return false
+    try {
+      const deviceToken = await this.getDeviceToken({})
+      await this.getXSTSToken({ userToken: ut.token, deviceToken }, { relyingParty })
```

Comment:
Not refreshing the XSTS token with correct device/title tokens is a good catch. However this doesn't account for the options passed by the user. Maybe adding this data to the cache would be a good idea.

Reply:
Yes, or if that's too complex passing down options and just returning false if options.authTitle is set will cause the correct handling code to be run